packages/core/src/query/flux.ts

```diff
@@ -0,0 +1,245 @@
+  for (const c of val) {
+    if (c === '.') {
+      if (dot) throw new Error(`not a flux float: ${val}`)
+      dot = !dot
+    }
+    if (c !== '.' && (c < '0' || c > '9'))
+      throw new Error(`not a flux float: ${val}`)
+  }
+  return val
+}
+/**
+ * Creates a flux float literal.
+ */
+export function fluxFloat(value: any): FluxParameterLike {
+  return new FluxParameter(sanitizeFloat(value))
+}
+
+function sanitizeDateTime(value: any): string {
+  return `time(v: "${sanitizeString(value)}")`
+}
+
+/**
+ * Creates flux date-time literal.
+ */
+export function fluxDateTime(value: any): FluxParameterLike {
+  return new FluxParameter(sanitizeDateTime(value))
+}
+
+/**
+ * Creates flux date-time literal.
+ */
+export function fluxDuration(value: any): FluxParameterLike {
+  return new FluxParameter(`duration(v: "${sanitizeString(value)}")`)
+}
+
+function sanitizeRegExp(value: any): string {
+  return `regexp.compile(v: "${sanitizeString(value)}")`
+}
+
+/**
+ * Creates flux regexp literal.
+ */
+export function fluxRegExp(value: any): FluxParameterLike {
+  // let the server decide if it can be parsed
+  return new FluxParameter(sanitizeRegExp(value))
+}
+
+/**
+ * Creates flux boolean literal.
+ */
+export function fluxBool(value: any): FluxParameterLike {
+  if (value === 'true' || value === 'false') {
+    return new FluxParameter(value)
+  }
+  return new FluxParameter((!!value).toString())
+}
+
+/**
+ * Assumes that the supplied string is a flux literal value and creates
+ * a parameter out of it.
+ */
+export function fluxLiteral(value: string): FluxParameterLike {
+  // let the server decide if it can be parsed
+  return new FluxParameter(value)
+}
+
+/**
+ * Escapes content of the supplied parameter so that it can be safely embedded into flux query.
+ * @param value parameter value
+ * @return sanitized flux value or an empty string if it cannot be converted
+ */
+export function toFluxValue(value: any): string {
+  if (value === undefined) {
+    return ''
+  } else if (value === null) {
+    return 'null'
+  } else if (typeof value === 'boolean') {
+    return value.toString()
+  } else if (typeof value === 'string') {
+    return `"${sanitizeString(value)}"`
+  } else if (typeof value === 'number') {
+    return sanitizeFloat(value)
+  } else if (
+    typeof value === 'object' &&
+    typeof value[FLUX_VALUE] === 'function'
+  ) {
+    return value[FLUX_VALUE]()
+  } else if (value instanceof Date) {
+    return value.toISOString()
+  } else if (value instanceof RegExp) {
+    return sanitizeRegExp(value)
+  }
+  return toFluxValue(value.toString())
+}
+
+/**
+ * Flux is a tagged template that sanitizes supplied parameters
+ * to avoid injection attacks in flux.
+ */
+export function flux(
+  strings: TemplateStringsArray,
+  ...values: any
+): ParameterizedQuery {
+  if (strings.length == 1) return strings[0] // the simplest case
+  const parts = new Array<string>(strings.length + values.length)
+  let partIndex = 0
```

Comment:
nit: whats the reason to initialize the array to a certain length and running an index variable, insteadof
```ts
const parts: string[] = [];
// ...
parts.push(part);
```
which is much more idiomatic code?

Reply:
since the size of the array is known in advance, I preferred this way